### PR TITLE
Reduce Bimi priority (it should only be displayed if we couldn't find an avatar)

### PR DESCRIPTION
**Pull request by KevinBoulongne**
When we received a Bimi, we displayed it automatically.
But we should only do that if we can't find an avatar.

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/views/AvatarView.kt
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/views/AvatarView.kt:1]: The logic in this class is already complicated enough. When supporting the bimi, I think we should avoid separating the places that choose if we display a bimi or an avatar to multiple places and we should also avoid letting the different `loadAvatar` methods handle too much concerns, as much as possible

Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/views/AvatarView.kt:1]: I'm trying some things out in this regard

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/views/AvatarView.kt
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/views/AvatarView.kt:1]: I ended up changing too many things for a code proposal so here is the kind of modifications I can think about: https://github.com/Infomaniak/android-kMail/pull/1941

File: app/src/main/java/com/infomaniak/mail/views/AvatarView.kt [hunk 1]
```diff
@@ -58,16 +58,24 @@ class AvatarView @JvmOverloads constructor(
...
+        val displayType = getAvatarDisplayType(currentCorrespondent, currentBimi)
+
+        if (displayType == AvatarDisplayType.CUSTOM_AVATAR || displayType == AvatarDisplayType.INITIALS) {
+            loadAvatarUsingDictionary(currentCorrespondent!!, contacts)
         }
     }
 
     @Inject
-    lateinit var avatarMergedContactData: AvatarMergedContactData
+    lateinit var _avatarMergedContactData: AvatarMergedContactData
```
Comment (NicolasBourdin88) [inline app/src/main/java/com/infomaniak/mail/views/AvatarView.kt:72]: Why add an underscore

**Review by LunarX — APPROVED** [review]

**Review by NicolasBourdin88 — APPROVED** [review]

**Pull request merged**